src/lib/Event/Listener/LoginListener.php

```diff
@@ -100,21 +113,58 @@ public function onSecurityInteractiveLogin(InteractiveLoginEvent $event): void
         }
     }
 
+    private function getCustomerId(string $siteAccessName): ?int
+    {
+        $parameterNameCustomerId = 'authentication.customer_id';
+
+        if (!$this->configResolver->hasParameter(
+            $parameterNameCustomerId,
+            Parameters::NAMESPACE,
+            $siteAccessName
+        )) {
+            return null;
+        }
+
+        $customerId = (int)$this->configResolver->getParameter(
+            $parameterNameCustomerId,
+            Parameters::NAMESPACE,
+            $siteAccessName
+        );
+
+        return $customerId === 0 ? null : $customerId;
```

Comment:
It could be simplified to: 

```suggestion
        return $customerId ?: null;
```